src/biogeophys/SnowSnicarMod.F90

```diff
@@ -90,54 +84,58 @@ module SnowSnicarMod
   ! (idx_Mie_snw_mx is number of snow radii with defined parameters (i.e. from 30um to 1500um))
   
   ! direct-beam weighted ice optical properties
-  real(r8) :: ss_alb_snw_drc(idx_Mie_snw_mx,numrad_snw)
-  real(r8) :: asm_prm_snw_drc(idx_Mie_snw_mx,numrad_snw)
-  real(r8) :: ext_cff_mss_snw_drc(idx_Mie_snw_mx,numrad_snw)
+  real(r8), pointer :: ss_alb_snw_drc(:,:) !(idx_Mie_snw_mx,numrad_snw)
+  real(r8), pointer :: asm_prm_snw_drc(:,:) !(idx_Mie_snw_mx,numrad_snw)
+  real(r8), pointer :: ext_cff_mss_snw_drc(:,:) !(idx_Mie_snw_mx,numrad_snw)
 
   ! diffuse radiation weighted ice optical properties
-  real(r8) :: ss_alb_snw_dfs(idx_Mie_snw_mx,numrad_snw)
-  real(r8) :: asm_prm_snw_dfs(idx_Mie_snw_mx,numrad_snw)
-  real(r8) :: ext_cff_mss_snw_dfs(idx_Mie_snw_mx,numrad_snw)
+  real(r8), pointer :: ss_alb_snw_dfs(:,:) !(idx_Mie_snw_mx,numrad_snw)
+  real(r8), pointer :: asm_prm_snw_dfs(:,:) !(idx_Mie_snw_mx,numrad_snw)
+  real(r8), pointer :: ext_cff_mss_snw_dfs(:,:) !(idx_Mie_snw_mx,numrad_snw)
 
   ! hydrophiliic BC
-  real(r8) :: ss_alb_bc1(numrad_snw)
-  real(r8) :: asm_prm_bc1(numrad_snw)
-  real(r8) :: ext_cff_mss_bc1(numrad_snw)
+  real(r8), pointer :: ss_alb_bc1(:) !(numrad_snw)
+  real(r8), pointer :: asm_prm_bc1(:) !(numrad_snw)
+  real(r8), pointer :: ext_cff_mss_bc1(:) !(numrad_snw)
 
   ! hydrophobic BC
-  real(r8) :: ss_alb_bc2(numrad_snw)
-  real(r8) :: asm_prm_bc2(numrad_snw)
-  real(r8) :: ext_cff_mss_bc2(numrad_snw)
+  real(r8), pointer :: ss_alb_bc2(:) !(numrad_snw)
+  real(r8), pointer :: asm_prm_bc2(:) !(numrad_snw)
+  real(r8), pointer :: ext_cff_mss_bc2(:) !(numrad_snw)
 
   ! hydrophobic OC
-  real(r8) :: ss_alb_oc1(numrad_snw)
-  real(r8) :: asm_prm_oc1(numrad_snw)
-  real(r8) :: ext_cff_mss_oc1(numrad_snw)
+  real(r8), pointer :: ss_alb_oc1(:) !(numrad_snw)
+  real(r8), pointer :: asm_prm_oc1(:) !(numrad_snw)
+  real(r8), pointer :: ext_cff_mss_oc1(:) !(numrad_snw)
```

Comment:
@cenlinhe I didn't notice this before, but hydrophilic BC has index 1, while hydrophilic OC has index 2. If there is no reason, then I would rather clean that up.

Reply:
This is an error in the comment not the variable. oc1 should be hydrophilic and oc2 should be hydrophobic. I checked the later code by assigning these values which actually is correct (oc1 is hydrophilic, oc2 is hydrophobic):

                  ! aerosol species 3 optical properties, hydrophilic OC
                  ss_alb_aer_lcl(3)        = ss_alb_oc1(bnd_idx)      
                  asm_prm_aer_lcl(3)       = asm_prm_oc1(bnd_idx)
                  ext_cff_mss_aer_lcl(3)   = ext_cff_mss_oc1(bnd_idx)

                  ! aerosol species 4 optical properties, hydrophobic OC
                  ss_alb_aer_lcl(4)        = ss_alb_oc2(bnd_idx)      
                  asm_prm_aer_lcl(4)       = asm_prm_oc2(bnd_idx)
                  ext_cff_mss_aer_lcl(4)   = ext_cff_mss_oc2(bnd_idx)